### ISSUE TITLE
ServiceBinder AndroidJavaException

### ISSUE THREAD

**Issue by SmartDeveloper29** [KirillKuzyk/Google-Play-License-Check#19]
I am getting the following exception on Android device

12/02 16:25:43.808 7439 7466 Error Unity AndroidJavaException: java.lang.ClassNotFoundException: com.unity3d.plugin.lvl.ServiceBinder
12/02 16:25:43.808 7439 7466 Error Unity java.lang.ClassNotFoundException: com.unity3d.plugin.lvl.ServiceBinder
12/02 16:25:43.808 7439 7466 Error Unity 	at java.lang.Class.classForName(Native Method)
12/02 16:25:43.808 7439 7466 Error Unity 	at java.lang.Class.forName(Class.java:324)
12/02 16:25:43.808 7439 7466 Error Unity 	at com.unity3d.player.UnityPlayer.nativeRender(Native Method)
12/02 16:25:43.808 7439 7466 Error Unity 	at com.unity3d.player.UnityPlayer.x(SourceFile)
12/02 16:25:43.808 7439 7466 Error Unity 	at com.unity3d.player.x0.handleMessage(SourceFile)
12/02 16:25:43.808 7439 7466 Error Unity 	at android.os.Handler.dispatchMessage(Handler.java:98)
12/02 16:25:43.808 7439 7466 Error Unity 	at android.os.Looper.loop(Looper.java:158)
12/02 16:25:43.808 7439 7466 Error Unity 	at com.unity3d.player.y0.run(SourceFile)
12/02 16:25:43.808 7439 7466 Error Unity Caused by: java.lang.ClassNotFoundException: Didn't find class "com.unity3d.plugin.lvl.ServiceBinder" on path: DexPathList[[zip file

**Comment by KyryloKuzyk** [comment]
Thank you for the bug report! I have an idea of what can cause the issue.

Gradle was updated in Unity [2022.3.38f1+](https://docs.unity3d.com/2022.3/Documentation/Manual/android-gradle-overview.html) and I guess that R8 is enabled by default now. R8 performs code stripping that removes the `com.unity3d.plugin.lvl.ServiceBinder` class from the build which results in the error you're seeing.

Can you please try the following?
1. Enable this setting: `Project Settings / Player / Android / Publishing Settings / Build / Custom Proguard File`.
<img width="592" alt="image" src="https://github.com/user-attachments/assets/967fcf48-7871-4168-8e5c-2c9b141727d8">  

2. Add this to the created `proguard-user.txt` file:
```
# Keep the ServiceBinder class and all its members
-keep class com.unity3d.plugin.lvl.ServiceBinder {
    <init>(...);
    *;
}
```

Please tell me If it works and I'll automate this setup process in the next version of the plugin.

**Comment by SmartDeveloper29** [comment]
Hi,



Thanks a lot for your help. That solved my problem.

**Comment by KyryloKuzyk** [comment]
I released an official fix in version 1.2.4. Huge thanks for reporting the issue!
